From: Lukas Lipka <lukas@pmad.net>

Small cleanup for converting a string to a char array

Index: Index/TermBuffer.cs
===================================================================
--- Index/TermBuffer.cs	(revision 4506)
+++ Index/TermBuffer.cs	(working copy)
@@ -86,14 +86,8 @@
 			
 			// copy text into the buffer
 			SetTextLength(term.Text().Length);
+			text = term.Text().ToCharArray();
 
-			System.String sourceString = term.Text();
-			int sourceEnd = term.Text().Length;
-			for (int i = 0; i < sourceEnd; i++)
-			{
-				text[i] = (char) sourceString[i];
-			}
-			
 			this.field = term.Field();
 			this.term = term;
 		}
@@ -143,4 +137,4 @@
 			return clone;
 		}
 	}
-}
\ No newline at end of file
+}
