Index: Index/SegmentInfos.cs
===================================================================
--- Index/SegmentInfos.cs	(revision 4083)
+++ Index/SegmentInfos.cs	(working copy)
@@ -150,5 +150,26 @@
 			sis.Read(directory);
 			return sis.GetVersion();
 		}
+
+		public void Optimize(Directory directory)
+		{
+			string[] files = directory.List();
+
+			System.Collections.ArrayList segment_names = new System.Collections.ArrayList();
+			foreach (SegmentInfo si in this)
+				segment_names.Add (si.name);
+
+			foreach (string file in files) {
+				string basename = System.IO.Path.GetFileNameWithoutExtension (file);
+				if (segment_names.Contains (basename))
+					continue;
+
+				if (basename == IndexFileNames.DELETABLE || basename == IndexFileNames.SEGMENTS)
+					continue;
+
+				Console.WriteLine ("WARNING! Deleting stale data {0}", file);
+				directory.DeleteFile (file);
+			}
+		}
 	}
-}
\ No newline at end of file
+}
Index: Index/IndexWriter.cs
===================================================================
--- Index/IndexWriter.cs	(revision 4083)
+++ Index/IndexWriter.cs	(working copy)
@@ -694,6 +694,8 @@
 					int minSegment = segmentInfos.Count - mergeFactor;
 					MergeSegments(minSegment < 0?0:minSegment);
 				}
+
+				segmentInfos.Optimize (directory);
 			}
 		}
 		
@@ -1001,4 +1003,4 @@
 			directory.RenameFile("deleteable.new", IndexFileNames.DELETABLE);
 		}
 	}
-}
\ No newline at end of file
+}
