From: Joe Shaw <joeshaw@novell.com>

Use UTC times

--- Lucene.Net/Store/FSDirectory.cs.orig	2005-10-02 09:56:40.000000000 +0100
+++ Lucene.Net/Store/FSDirectory.cs	2005-10-02 09:58:05.000000000 +0100
@@ -405,21 +405,21 @@ namespace Lucene.Net.Store
 		public override long FileModified(System.String name)
 		{
 			System.IO.FileInfo file = new System.IO.FileInfo(System.IO.Path.Combine(directory.FullName, name));
-			return (file.LastWriteTime.Ticks);
+			return (file.LastWriteTimeUtc.Ticks);
 		}
 		
 		/// <summary>Returns the time the named file was last modified. </summary>
 		public static long FileModified(System.IO.FileInfo directory, System.String name)
 		{
 			System.IO.FileInfo file = new System.IO.FileInfo(System.IO.Path.Combine(directory.FullName, name));
-			return (file.LastWriteTime.Ticks);
+			return (file.LastWriteTimeUtc.Ticks);
 		}
 		
 		/// <summary>Set the modified time of an existing file to now. </summary>
 		public override void  TouchFile(System.String name)
 		{
 			System.IO.FileInfo file = new System.IO.FileInfo(System.IO.Path.Combine(directory.FullName, name));
-            file.LastWriteTime = System.DateTime.Now;
+            file.LastWriteTime = System.DateTime.UtcNow;
 		}
 		
 		/// <summary>Returns the length in bytes of a file in the directory. </summary>
--- Lucene.Net/Store/RAMDirectory.cs.orig	2005-10-02 11:13:09.000000000 +0100
+++ Lucene.Net/Store/RAMDirectory.cs	2005-10-02 11:13:26.000000000 +0100
@@ -161,7 +161,7 @@ namespace Lucene.Net.Store
 			//     final boolean MONITOR = false;
 			
 			RAMFile file = (RAMFile) files[name];
-			long ts2, ts1 = System.DateTime.Now.Ticks;
+			long ts2, ts1 = System.DateTime.UtcNow.Ticks;
 			do 
 			{
 				try
@@ -171,7 +171,7 @@ namespace Lucene.Net.Store
 				catch (System.Threading.ThreadInterruptedException)
 				{
 				}
-				ts2 = System.DateTime.Now.Ticks;
+				ts2 = System.DateTime.UtcNow.Ticks;
 				//       if (MONITOR) {
 				//         count++;
 				//       }
@@ -235,4 +235,4 @@ namespace Lucene.Net.Store
 		{
 		}
 	}
-}
\ No newline at end of file
+}
--- Lucene.Net/Store/RAMFile.cs.orig	2005-10-02 11:14:25.000000000 +0100
+++ Lucene.Net/Store/RAMFile.cs	2005-10-02 11:14:27.000000000 +0100
@@ -21,6 +21,6 @@ namespace Lucene.Net.Store
 	{
 		internal System.Collections.ArrayList buffers = System.Collections.ArrayList.Synchronized(new System.Collections.ArrayList(10));
 		internal long length;
-		internal long lastModified = System.DateTime.Now.Ticks;
+		internal long lastModified = System.DateTime.UtcNow.Ticks;
 	}
-}
\ No newline at end of file
+}
--- Lucene.Net/Store/RAMOutputStream.cs.orig	2005-10-02 11:15:25.000000000 +0100
+++ Lucene.Net/Store/RAMOutputStream.cs	2005-10-02 11:15:35.000000000 +0100
@@ -103,7 +103,7 @@ namespace Lucene.Net.Store
 			if (pointer > file.length)
 				file.length = pointer;
 			
-			file.lastModified = System.DateTime.Now.Ticks;
+			file.lastModified = System.DateTime.UtcNow.Ticks;
 		}
 		
 		public override void  Close()
@@ -121,4 +121,4 @@ namespace Lucene.Net.Store
 			return file.length;
 		}
 	}
-}
\ No newline at end of file
+}
