From: Jon Trowbridge <trow@novell.com>

Open the FileStream with FileShare.ReadWrite.

--- Lucene.Net/Store/FSDirectory.cs.orig	2005-10-02 10:11:34.000000000 +0100
+++ Lucene.Net/Store/FSDirectory.cs	2005-10-02 10:12:28.000000000 +0100
@@ -500,7 +500,7 @@ namespace Lucene.Net.Store
                     try
                     {
                         in_Renamed = new System.IO.BinaryReader(System.IO.File.Open(old.FullName, System.IO.FileMode.Open, System.IO.FileAccess.Read));
-                        out_Renamed = new System.IO.FileStream(nu.FullName, System.IO.FileMode.Create);
+                        out_Renamed = new System.IO.FileStream(nu.FullName, System.IO.FileMode.Create, System.IO.FileAccess.Write, System.IO.FileShare.ReadWrite);
                         // see if the buffer needs to be initialized. Initialization is
                         // only done on-demand since many VM's will never run into the renameTo
                         // bug and hence shouldn't waste 1K of mem for no reason.
