Index: Store/FSDirectory.cs
===================================================================
--- Store/FSDirectory.cs	(revision 3170)
+++ Store/FSDirectory.cs	(working copy)
@@ -133,29 +133,20 @@
 			{
 				if (Lucene.Net.Store.FSDirectory.disableLocks || Enclosing_Instance.InstanceDisableLock)
 					return ;
-				bool tmpBool;
-				if (System.IO.File.Exists(lockFile.FullName))
-				{
-					System.IO.File.Delete(lockFile.FullName);
-					tmpBool = true;
-				}
-				else if (System.IO.Directory.Exists(lockFile.FullName))
-				{
-					System.IO.Directory.Delete(lockFile.FullName);
-					tmpBool = true;
-				}
-				else
-					tmpBool = false;
+
+				int fd = Mono.Unix.Native.Syscall.unlink (
+		    		    	lockFile.FullName);
+		    		if (fd == -1)
+					throw new System.IO.IOException (
+						"Could not release lock file: "
+						+ Mono.Unix.Native.Stdlib.strerror (Mono.Unix.Native.Stdlib.GetLastError ()
+					));
+
 				if (System.IO.File.Exists(lockFile.FullName)) {
-					Log ("Release didnt delete lockfile {0}.", lockFile.FullName);
-					tmpBool = false;
+					Beagle.Util.Logger.Log.Warn ("Release didnt delete lockfile {0}.", lockFile.FullName);
 				}
-				bool generatedAux = tmpBool;
-				if (tmpBool)
-					Log ("Released lock {0}", lockFile.FullName);
-				else
-					Log ("Failed to release lock {0}", lockFile.FullName);
 			}
+
 			public override bool IsLocked()
 			{
 				if (Lucene.Net.Store.FSDirectory.disableLocks || Enclosing_Instance.InstanceDisableLock)
